keps/sig-scheduling/5710-workload-aware-preemption/README.md

```diff
@@ -0,0 +1,1064 @@
+- go over the remaining potential victims starting from the highest priority and check if these can
+  be placed in the place they are currently running; if so remove from the potential victims
+
+As a bonus we may consider few potential placements of the new workload W here and choose the one that
+somehow optimizes the number of victims. But that will become more critical once we get to
+Topology-Aware-Scheduling and I would leave that optimization until then.
+<<[/UNRESOLVED]>>
+```
+
+```
+<<[UNRESOLVED sharing algorithms]>>
+The remaining question is to what extent we want to unify the preemption mechanism across
+pod-triggerred (existing algorithm) and workload-triggerred preemption.
+
+It might be tempting to start with a dedicated new implementation to reduce the risk. But the above
+proposal was structured such way to facilitate sharing:
+- once the new workload W is placed, going over the remaining potential victims and trying to
+  place them where they are currently running, is pretty much exactly what the current algorithm is
+  doing
+- considering "few potential placements" in the pod-triggerred case can be used as "try every node"
+  so effectively it's also the existing algorithm (just viewed from a slightly different angle
+
+So I would actually argue to we should refactor the existing preemption code and use that in both
+cases.
+<<[/UNRESOLVED]
+```
+
+### Delayed preemption
+
+```
+<<[UNRESOLVED delayed preemption]>>
+Should we leave it as part of this KEP or should this be moved to the Gang-Scheduling one?
+<<[/UNRESOLVED]>>
+```
+
+As part of minimizing preemptions goal, arguably the most important thing to do is to avoid unnecessary
+preemptions. However, this is not true for the current gang scheduling implementation.
+In the current implementation, preemption is triggered in the `PostFiler`. However, it's entirely
+possible that a given pod may actually not even proceed to binding, because we can't schedule the
+whole gang. In such case, the preemption ended up being a completely unnecessary disruption.
+
+We will address it with what we call `delayed preemption` mechanism as following:
+
+1. We will modify the `DefaultPreemption` plugin to just compute preemptions, without actuating those.
+   At this point these should only be stored in kube-scheduler's memory.
+
+```
+<<[UNRESOLVED storing victims]>>
+Decide how exactly victims should be stored. Potential options:
+1. New field in the pod like `pod.Spec.PreemptionTargets` if it gets introduces by
+   [Kubernetes Scheduling Races Handling] (without writing it to kube-apiserver).
+1. New field in the workload object (delayed preemption will not bring much value in
+   case of scheduling individual pods, though there would be significant benefit from
+   unification, so probably this isn't ideal option).
+1. Storing it in private kube-scheduler' structures (PodInfo for individual pods and
```

Comment:
This does not allow external schedulers to use the same concept for victims nomination.